From f4bbe3531fae6523c582b55052448078e0a63f9f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Amadeusz=20S=C5=82awi=C5=84ski?=
 <amadeuszx.slawinski@intel.com>
Date: Fri, 9 Nov 2018 16:08:27 +0100
Subject: [PATCH 235/292] ASoC: Intel: Skylake: Add function to cleanup debugfs
 interface
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Currently debugfs has no cleanup function. Add skl_debufs_exit function
so we can clean after ourselves properly.

Change-Id: I3d9e74964880b3ec602ecfc3035e02ca878cf509
Signed-off-by: Amadeusz Sławiński <amadeuszx.slawinski@intel.com>
Reviewed-on:
---
 sound/soc/intel/skylake/skl-debug.c | 8 ++++++++
 sound/soc/intel/skylake/skl.h       | 3 +++
 2 files changed, 11 insertions(+)

diff --git a/sound/soc/intel/skylake/skl-debug.c b/sound/soc/intel/skylake/skl-debug.c
index 69085ac0b9d9..bd340970b105 100644
--- a/sound/soc/intel/skylake/skl-debug.c
+++ b/sound/soc/intel/skylake/skl-debug.c
@@ -959,3 +959,11 @@ struct skl_debug *skl_debugfs_init(struct skl *skl)
 	debugfs_remove_recursive(d->fs);
 	return NULL;
 }
+
+void skl_debugfs_exit(struct skl *skl)
+{
+	struct skl_debug *d = skl->debugfs;
+
+	skl_exit_nhlt(d);
+	debugfs_remove_recursive(d->fs);
+}
diff --git a/sound/soc/intel/skylake/skl.h b/sound/soc/intel/skylake/skl.h
index 12a72a7a47f6..c64ff3dfbe1b 100644
--- a/sound/soc/intel/skylake/skl.h
+++ b/sound/soc/intel/skylake/skl.h
@@ -240,6 +240,7 @@ int skl_platform_pcm_trigger(struct snd_pcm_substream *substream,
 
 #ifdef CONFIG_DEBUG_FS
 struct skl_debug *skl_debugfs_init(struct skl *skl);
+void skl_debugfs_exit(struct skl *skl);
 void skl_debug_init_module(struct skl_debug *d,
 			struct snd_soc_dapm_widget *w,
 			struct skl_module_cfg *mconfig);
@@ -252,6 +253,8 @@ static inline struct skl_debug *skl_debugfs_init(struct skl *skl)
 {
 	return NULL;
 }
+static inline void skl_debugfs_exit(struct skl *skl)
+{}
 static inline void skl_debug_init_module(struct skl_debug *d,
 					 struct snd_soc_dapm_widget *w,
 					 struct skl_module_cfg *mconfig)
-- 
2.21.0

